enso-org/enso
# Refactor dashboard and settings tabs into routes

### Pull Request Description
- Close https://github.com/enso-org/cloud-v2/issues/1410
  - Enables links in the future to link directory to e.g. a specific settings page
    - Deep links *do not* support these new links yet, however it would be simple enough to add support
  - Allow "settings" icon in Category Switcher to do `navigate('/settings/local')` rather than requiring a workaround
  - Remove obsolete `searchParamsState` hooks

### Important Notes
- When navigating, the URL search parameters are lost (and so they are missing by the time GUI reads them).
  - GUI team should review whether this is an issue for the time being
  - We should probably rethink how CLI parameters should be passed to GUI (if at all)

### Testing instructions
Apart from testing all pages (all settings pages, all drive categories, editor tab):
- Make sure you stay on the same page (including settings subpages and drive categories) after a reload
- Test opening and closing projects and switching tabs with projects open
- Make sure that projects are still reopened after reload (and do not magically appear after reload if closed
- Make sure settings tab shows as selected after reload (because the active tab is done by adding a clip mask to the darker background behind it, so if there is a race condition then it may get out of sync)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
